src/libsystemd-network/network-internal.c

```diff
@@ -40,15 +40,18 @@ const char *net_get_name(sd_device *device) {
 
 int net_get_unique_predictable_data(sd_device *device, uint64_t *result) {
         size_t l, sz = 0;
-        const char *name = NULL;
+        const char *name;
         int r;
         uint8_t *v;
 
         assert(device);
 
         name = net_get_name(device);
         if (!name)
-                return -ENOENT;
+                if (sd_device_get_property_value(device, "ID_BUS", NULL) == -ENOENT)
+                        (void) sd_device_get_sysname(device, &name);
+        if (!name)
+                return -ENODATA;
```

Comment:
Can we add some explanation here in a comment what this is doing, and why? Also, I think it would be a great idea if we'd have debug logging here about what we are doing here, since I am pretty sure it's highly interesting to figure out sometimes how a specific MAC came to be...